### PR TITLE
Add Pylint support and enable Pylint CI checks

### PR DESCRIPTION
**Pull request by correctmost**
## PR Description:

Pylint can help catch issues like #2625:

```
************* Module archinstall.default_profiles.server
archinstall/default_profiles/server.py:34:4: E1101: Instance of 'ServerProfile' has no 'set_current_selection' member (no-member)
```

I didn't add the Pylint checks to the pre-commit config because they might be a bit too slow to run on each commit.

## Tests and Checks
- [ ] I have tested the code!<br>

### CODE REVIEW

**Comment by svartkanin** [comment]
Maybe I'm missing something here but why is there a need for ruff and pylint? Isn't ruff covering everything?

**Comment by Torxed** [comment]
Before we merge this, I have to create a `python-pylint-pydantic` package as we need to be able to source this during [PKGBUILD](https://github.com/archlinux/archinstall/blob/master/PKGBUILD):
```diff
makedepends=(
  'python-setuptools'
  'python-sphinx'
  'python-build'
  'python-installer'
  'python-wheel'
+ 'python-pylint'
+ 'python-pylint-pydantic'
)
```
However that package does not exist currently in [any repo](https://archlinux.org/packages/?sort=&q=pylint&maintainer=&flagged=), not even in [AUR](https://aur.archlinux.org/packages?O=0&K=pylint).

It's not a major showstopper, but I just need to sit down a few minutes and make that happen.

**Comment by svartkanin** [comment]
@Torxed why is it needed for a build? Those packages are part of the dev packages in pyproject.toml so they shouldn't be dependencies for a build?

**Comment by Torxed** [comment]
Because we *should* run [check()](https://wiki.archlinux.org/title/Python_package_guidelines#Check) in [PKGBUILD](https://wiki.archlinux.org/title/PKGBUILD#checkdepends).
I haven't been a model packaging citizen, but now that we're doing good stuff this should ideally be there :)

**Comment by svartkanin** [comment]
Hmm I'm still somewhat worried that mixing the build with linting may cause issues down the line. If these checks run on PRs and only PRs that validate successfully get merged then the master branch should be considered ckean and ready for a build.
I remember @Torxed in the past rare PRs got merged with failing checks coz fixing a bug was higher priority then meeting linting checks. If such a check is now failing and the PR gets merged it'll block a hit fix release completely.

**Comment by Torxed** [comment]
> Hmm I'm still somewhat worried that mixing the build with linting may cause issues down the line. If these checks run on PRs and only PRs that validate successfully get merged then the master branch should be considered ckean and ready for a build. I remember @Torxed in the past rare PRs got merged with failing checks coz fixing a bug was higher priority then meeting linting checks. If such a check is now failing and the PR gets merged it'll block a hit fix release completely.

There's also the odd "arch packaging patches" where any trusted user can go in and submit `-<arch patch>` versions. 

Which is what you see sometimes where, a package didn't change version upstream like `v2.8.6` might be packaged with `v2.8.6-2` only in arch. 

It's a low probability of issues, but best practice in packaging is to have the `check()` hook during packaging that should just run a final test/linting.

And it's no problem for me to make a PKGBUILD for pylint-pydantic as long as it's maintained upstream :)

I just need to find my way to my laptop 😅

**Comment by svartkanin** [comment]
Maybe we should pause the pylint addition for a bit and see how well ruff in combination with the others work, get mypy fully compliant and see if we need pylint in addition?

**Comment by Torxed** [comment]
It's true, we could skip the `check()` in PKGBUILD's for now.
But knowing myself it will probably be forgotten until someone pokes me or when it bites me in the butt.

But the tests that pylint adds sounds nice tbh.

**Comment by svartkanin** [comment]
@correctmost I'm soon (TM) about to raise an integration PR for the new menu system which is going to shuffle things around, so just to prevent any throwaway work being done just ignore any menu related linting

**Comment by svartkanin** [comment]
https://github.com/archlinux/archinstall/pull/2663

**Comment by Torxed** [comment]
I've successfully built the dependencies now. I just need to sign and publish the packages :)

**Review by Torxed — APPROVED** [review]

**Pull request merged**
